src/bit-systems/object-menu.ts

```diff
@@ -39,10 +52,69 @@ function moveToTarget(world: HubsWorld, menu: EntityID) {
   targetObj.updateMatrices();
 
+  // TODO: position the menu more carefully...
+  //       For example, if a menu object is just placed at a target
+  //       object's position the menu object can be hidden by a large
+  //       target object or the menu object looks too small for a far
+  //       target object.
+  _mat4.copy(targetObj.matrixWorld);
+
+  // Keeps world scale (1, 1, 1) because
+  // a menu object is a child of a target object
+  // and the target object's scale can be changed.
+  // Another option may be making the menu object
+  // a sibling of the target object.
+  _mat4.decompose(_vec3_1, _quat, _vec3_2);
+  _vec3_2.set(1.0, 1.0, 1.0);
+  _mat4.compose(_vec3_1, _quat, _vec3_2);
+
   const menuObj = world.eid2obj.get(menu)!;
+  setMatrixWorld(menuObj, _mat4);
 
-  // TODO: position the menu more carefully...
-  setMatrixWorld(menuObj, targetObj.matrixWorld);
+  // TODO: Remove the dependency with AFRAME
+  const camera = AFRAME.scenes[0].systems["hubs-systems"].cameraSystem.viewingCamera;
+  camera.updateMatrices();
+  menuObj.lookAt(_vec3_1.setFromMatrixPosition(camera.matrixWorld));
+}
+
+// TODO: startRotation/Scaling() and stopRotation/Scaling() are
+//       temporary implementation that rely on the old systems.
+//       They should be rewritten more elegantly with bitecs.
+function startRotation(world: HubsWorld, targetEid: EntityID) {
+  // TODO: Don't use any
+  // TODO: Remove the dependency with AFRAME
+  const transformSystem = (AFRAME as any).scenes[0].systems["transform-selected-object"];
+  const physicsSystem = AFRAME.scenes[0].systems["hubs-systems"].physicsSystem;
+  physicsSystem.updateBodyOptions(Rigidbody.bodyId[targetEid], { type: "kinematic" });
+  const rightCursorEid = anyEntityWith(world, RemoteRight)!;
+  transformSystem.startTransform(world.eid2obj.get(targetEid), world.eid2obj.get(rightCursorEid), {
+    mode: TRANSFORM_MODE.CURSOR
+  });
+}
+
+function stopRotation() {
+  // TODO: Don't use any
+  // TODO: Remove the dependency with AFRAME
+  const transformSystem = (AFRAME as any).scenes[0].systems["transform-selected-object"];
+  transformSystem.stopTransform();
+}
+
+function startScaling(world: HubsWorld, targetEid: EntityID) {
+  // TODO: Don't use any
+  // TODO: Remove the dependency with AFRAME
+  const transformSystem = (AFRAME as any).scenes[0].systems["transform-selected-object"];
+  const physicsSystem = AFRAME.scenes[0].systems["hubs-systems"].physicsSystem;
+  physicsSystem.updateBodyOptions(Rigidbody.bodyId[targetEid], { type: "kinematic" });
+  const rightCursorEid = anyEntityWith(world, RemoteRight)!;
+  scalingHandler = new ScalingHandler(world.eid2obj.get(targetEid), transformSystem);
+  scalingHandler!.objectToScale = world.eid2obj.get(targetEid);
+  scalingHandler!.startScaling(world.eid2obj.get(rightCursorEid));
+}
+
+function stopScaling(world: HubsWorld) {
+  const rightCursorEid = anyEntityWith(world, RemoteRight)!;
```

Comment:
Not a suggestion for here but reminds me of an observation I want us to think about. 

I find it annoying that we don't have a good way to statically define references like this and have to "look them up" every frame with a query like this. Its not particularly expensive or anything, just silly because we actually know what number entity this is going to be every time. The only thing really stopping us from doing it is module loading order.

Reply:
I don't love `anyEntityWith()`. Added this topic in the internal design document.